test/contracts/core/modules/rebalancingSetIssuanceModule.spec.ts

```diff
@@ -0,0 +1,874 @@
+
+contract('RebalancingSetIssuanceModule', accounts => {
+  const [
+    ownerAccount,
+    functionCaller,
+    whitelist,
+  ] = accounts;
+
+  let core: CoreContract;
+  let transferProxy: TransferProxyContract;
+  let vault: VaultContract;
+  let rebalancingSetTokenFactory: RebalancingSetTokenFactoryContract;
+  let setTokenFactory: SetTokenFactoryContract;
+  let rebalancingTokenIssuanceModule: RebalancingSetIssuanceModuleContract;
+  let wethMock: WethMockContract;
+
+  const coreWrapper = new CoreWrapper(ownerAccount, ownerAccount);
+  const erc20Wrapper = new ERC20Wrapper(ownerAccount);
+  const rebalancingWrapper = new RebalancingWrapper(
+    ownerAccount,
+    coreWrapper,
+    erc20Wrapper,
+    blockchain
+  );
+
+  before(async () => {
+    ABIDecoder.addABI(Core.abi);
+    ABIDecoder.addABI(RebalancingSetIssuanceModule.abi);
+
+    transferProxy = await coreWrapper.deployTransferProxyAsync();
+    vault = await coreWrapper.deployVaultAsync();
+    core = await coreWrapper.deployCoreAsync(transferProxy, vault);
+    wethMock = await erc20Wrapper.deployWrappedEtherAsync(ownerAccount);
+
+    setTokenFactory = await coreWrapper.deploySetTokenFactoryAsync(core.address);
+
+    await coreWrapper.setDefaultStateAndAuthorizationsAsync(core, vault, transferProxy, setTokenFactory);
+
+    rebalancingSetTokenFactory = await coreWrapper.deployRebalancingSetTokenFactoryAsync(core.address, whitelist);
+    await coreWrapper.addFactoryAsync(core, rebalancingSetTokenFactory);
+
+    rebalancingTokenIssuanceModule = await coreWrapper.deployRebalancingSetIssuanceModuleAsync(
+      core,
+      vault,
+      transferProxy,
+      wethMock,
+    );
+    await coreWrapper.addModuleAsync(core, rebalancingTokenIssuanceModule.address);
+
+  });
+
+  after(async () => {
+    ABIDecoder.removeABI(Core.abi);
+    ABIDecoder.removeABI(RebalancingSetIssuanceModule.abi);
+  });
+
+  beforeEach(async () => {
+    await blockchain.saveSnapshotAsync();
+  });
+
+  afterEach(async () => {
+    await blockchain.revertAsync();
+  });
+
+  describe('#issueRebalancingSet', async () => {
+    let subjectCaller: Address;
+    let subjectRebalancingSetAddress: Address;
+    let subjectRebalancingSetQuantity: BigNumber;
+    let subjectKeepChangeInVault: boolean;
+
+    let baseSetComponent: StandardTokenMockContract;
+    let baseSetToken: SetTokenContract;
+    let baseSetNaturalUnit: BigNumber;
+    let rebalancingSetToken: RebalancingSetTokenContract;
+    let rebalancingUnitShares: BigNumber;
+    let baseSetComponentUnit: BigNumber;
+    let baseSetIssueQuantity: BigNumber;
+
+    let customRebalancingUnitShares: BigNumber;
+    let customRebalancingSetQuantity: BigNumber;
+
+    beforeEach(async () => {
+      subjectCaller = functionCaller;
+
+      baseSetComponent = await erc20Wrapper.deployTokenAsync(subjectCaller);
+      await erc20Wrapper.approveTransferAsync(baseSetComponent, transferProxy.address, subjectCaller);
+
+      // Create the Set (1 component)
```

Comment:
Let's do two components since the logic between `issueRebalancingSet` and `issueRebalancingSetWithEther` don't flow through each other. Want to make sure all the for loops in the transaction are working properly.